### ISSUE TITLE
Notify about alert errors

### ISSUE THREAD

**Issue by grzkv** [bosun-monitor/bosun#2258]
Errors may happen during an alert lifecycle. They will be shown in Bosun. But, people that are subscribed to the alert will not get notified. The ones who maintain the alert will get no heads-up as well. This turns out to be a problem when many people collaborate on alerts setup.

As the result, alert maintainers will be unaware of errors. Errors will persist and accumulate. Even worse, if an accident happens, an alert will be expected, but it may be broken. Alert will never come, and the accident will go undiscovered.

Sample scenario: **A** sets data collection to Graphite path `my.data.metric` → **B** makes an alert on top of it, e.g. `crit my.data.metric < 100` → **A** changes path to `my.data.metric_updated` without notifying **B** → alert breaks → **B** is unaware. 

We suggest adding error notifications to address this problem.

There are a couple of possible solutions for that:

1. Add alert keyword `notifyOnErrors = true` (`false` by default). When set to `true`, errors will go through `warnNotification` route using a pre-defined or set template. This solution is the most simple and minimal, but not ideal. It can be the starting point.
2. Add alert keyword `errorNotification=errorNotificationDefinition`. When defined, errors will go through indicated notification route using a pre-defined or set template.

Please share your thoughts!

**Comment by grzkv** [comment]
@kylebrandt @captncraig Could you please share your opinion on this issue? Which route is the best to take? What details should be taken into account?

**Comment by mvuets** [comment]
In our set-up having this functionality would be highly appreciated. Our problem is that we do not own the rule config ourselves entirely, but let other people create their own alerts. When alerts break (they do, it stroke us pretty badly in the recent past) hardly anyone notices, because it requires proactive and regular checking in the Bosun web UI.

If the general idea is fine by @kylebrandt and @captncraig, our team is more than happy to implement the proposed functionality. Mm? (-:

**Comment by kylebrandt** [comment]
Short Version: I agree.

Context / History:

- "Errors" were added as thing primarily to deal with the various time series datasources (for us, elastic and TSDB) being fairly unreliable. So it was a way to take a bunch of alert noise and stick it "somewhere else"
 - I don't think the concept of what exactly "Errors" are in this sense was ever well defined 
 - When we did the template refactor (adding things like custom notifications for ack, close etc) we punted on this problem

Something like `errorNotification=errorNotificationDefinition` seems like the correct path to me. 

It will probably need a lot of tests, and we need to make we have good definition (or at least list of) as an "Error" in terms of design.
